pkg/access/rewrap.go

```diff
@@ -278,3 +279,179 @@ func (p *Provider) Handler(w http.ResponseWriter, r *http.Request) {
 		slog.ErrorContext(ctx, "rewrap: marshall response failed", "err", err)
 	}
 }
+
+func nanoTDFRewrap(requestBody RequestBody) ([]byte, error) {
+	header := requestBody.KeyAccess.Header
+
+	headerReader := bytes.NewReader(header)
+
+	nanoTDF, err := nanotdf.ReadNanoTDFHeader(headerReader)
+	if err != nil {
+		slog.Error("Could not fetch attribute definitions from attributes service!", "err", err)
+	}
+
+	x, y := elliptic.UnmarshalCompressed(elliptic.P256(), nanoTDF.EphemeralPublicKey.Key)
+	if x == nil || y == nil {
+		return nil, errors.New("failed to unmarshal ephemeral public key")
+	}
+
+	kasEcPrivKeyFilePath := os.Getenv("KAS_EC_SECP256R1_PRIVATE_KEY")
+
+	// Load PEM file
+	raw, err := os.ReadFile(kasEcPrivKeyFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read KAS private key JSON data: %w", err)
+	}
+
+	block, _ := pem.Decode(raw)
+	privateKey, err := parsePrivateKey(block.Bytes)
+	if err != nil {
+		slog.Error("Failed to encode private key to DER", "err", err)
+	}
+
+	symmetricKey, err := generateSymmetricKey(nanoTDF.EphemeralPublicKey.Key, privateKey.(*ecdsa.PrivateKey))
+	if err != nil {
+		slog.Error("Failed to generate symmetric key", "err", err)
+	}
+
+	// Generate a private key
+	privateKeyEphemeral, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		slog.Error("Failed to create ec pair:", "err", err)
```

Comment:
See above - this probalby should just panic

```suggestion
		return nil, fmt.Errorf("failed to generate private key: %w", err)
```